### PR TITLE
Add experimental support for Ubuntu focal

### LINKED ISSUE #248: Support for Ubuntu 20.04 LTS (Focal Fossa)
**Is your feature request related to a problem? Please describe.**
A task [setup-debian.yml](https://github.com/nginxinc/ansible-role-nginx/blob/master/tasks/amplify/setup-debian.yml) fails with the following exception

```
The full traceback is:
  File "/tmp/ansible_apt_repository_payload_4hefh4mw/ansible_apt_repository_payload.zip/ansible/modules/packaging/os/apt_repository.py", line 548, in main
  File "/usr/lib/python3/dist-packages/apt/cache.py", line 591, in update
    raise FetchFailedException(e)
fatal: [host]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "codename": null,
            "filename": "nginx-amplify",
            "install_python_apt": true,
            "mode": null,
            "repo": "deb [arch=amd64] http://packages.amplify.nginx.com/ubuntu/ focal amplify-agent",
            "state": "present",
            "update_cache": true,
            "validate_certs": true
        }
    },
    "msg": "apt cache update failed"
}
```

**Describe the solution you'd like**
Add packages for [_focal_](https://releases.ubuntu.com/20.04/) within http://packages.amplify.nginx.com/ubuntu/dists/

**Describe alternatives you've considered**
n/a

**Additional context**
There is a related issue https://github.com/nginxinc/nginx-amplify-agent/issues/177

### PR DESCRIPTION
**Pull request by alessfg**
### Proposed changes
Add experimental support for Ubuntu focal and work on some minor formatting improvements. Resolves #248. 

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

-   [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/ansible-role-nginx/blob/master/CONTRIBUTING.md) document
-   [x] If necessary, I have added Molecule tests that prove my fix is effective or that my feature works
-   [x] I have checked that all Molecule tests pass after adding my changes
-   [x] I have updated any necessary documentation (`defaults/main/*.yml`, `README.md` and `CHANGELOG.md`)

### CODE REVIEW

File: tasks/keys/setup-keys.yml [hunk 1]
```diff
@@ -4,12 +4,12 @@
   when: ansible_os_family == "Alpine"
   tags: nginx_apkkey
 
-- name: "(Setup: Keys) Debian"
+- name: "(Setup: Keys) Debian/Ubuntu"
   include_tasks: "{{ role_path }}/tasks/keys/apt-key.yml"
   when: ansible_os_family == "Debian"
```
Comment (damex) [inline tasks/keys/setup-keys.yml:9]: if you want to add 'Ubuntu' support in addition to 'Debian' - it is needed to add a contain that covers 'Ubuntu'

``` when: ansible_distribution == 'Debian' or ansible_distribution == 'Ubuntu' ```

Comment (alessfg) [reply tasks/keys/setup-keys.yml:9]: Ubuntu's `ansible_os_family` is Debian.

**Comment by TimeTravelersHackedMe** [comment]
@alessfg It looks like this is no longer working or there is a missing dependency.

Here's what I'm seeing:
 FAILED! => {"changed": false, "msg": "apt cache update failed"}

Full log here:

```
task path: /home/hawkwood/.ansible/roles/nginxinc.nginx/tasks/amplify/setup-debian.yml:2
<NUC> (0, b'ansible-tmp-1596602165.7738736-210641032471126=/tmp/ansible-tmp-1596602165.7738736-210641032471126\n', b'OpenSSH_8.2p1 Ubuntu-4ubuntu0.1, OpenSSL 1.1.1f  31 Mar 2020\r\ndebug1: Reading configuration data /home/hawkwood/.ssh/config\r\ndebug1: /home/hawkwood/.ssh/config line 1: Applying options for *\r\ndebug1: /home/hawkwood/.ssh/config line 20: Applying options for NUC\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no files\r\ndebug1: /etc/ssh/ssh_config line 21: Applying options for *\r\ndebug2: resolve_canonicalize: hostname *is address\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 1041\r\ndebug3: mux_client_request_session: session request sent\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
Using module file /usr/lib/python3/dist-packages/ansible/modules/packaging/os/apt_repository.py
<NUC> PUT /home/hawkwood/.ansible/tmp/ansible-local-10191hmcex92/tmp97jmgr8g TO /tmp/ansible-tmp-1596602165.7738736-210641032471126/AnsiballZ_apt_repository.py
<NUC> ESTABLISH SSH CONNECTION FOR USER: hawkwood
<NUC> SSH: ansible.cfg set ssh_args: (-C)(-o)(ControlMaster=auto)(-o)(ControlPersist=60s)
<NUC> SSH: ANSIBLE_HOST_KEY_CHECKING/host_key_checking disabled: (-o)(StrictHostKeyChecking=no)
<NUC> SSH: ANSIBLE_REMOTE_PORT/remote_port/ansible_port set: (-o)(Port=2214)
<NUC> SSH: ANSIBLE_REMOTE_USER/remote_user/ansible_user/user/-u set: (-o)(User="hawkwood")
<NUC> SSH: ANSIBLE_TIMEOUT/timeout set: (-o)(ConnectTimeout=10)
<NUC> SSH: PlayContext set ssh_common_args: ()
<NUC> SSH: PlayContext set ssh_extra_args: ()
<NUC> SSH: found only ControlPersist; added ControlPath: (-o)(ControlPath=/home/hawkwood/.ansible/cp/884cb4555b)
<NUC> SSH: EXEC sshpass -d11 ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=2214 -o 'User="hawkwood"' -o ConnectTimeout=10 -o ControlPath=/home/hawkwood/.ansible/cp/884cb4555b NUC 'dd of=/tmp/ansible-tmp-1596602165.7738736-210641032471126/AnsiballZ_apt_repository.py bs=65536'
<NUC> (0, b'', b'OpenSSH_8.2p1 Ubuntu-4ubuntu0.1, OpenSSL 1.1.1f  31 Mar 2020\r\ndebug1: Reading configuration data /home/hawkwood/.ssh/config\r\ndebug1: /home/hawkwood/.ssh/config line 1: Applying options for *\r\ndebug1: /home/hawkwood/.ssh/config line 20: Applying options for NUC\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no files\r\ndebug1: /etc/ssh/ssh_config line 21: Applying options for *\r\ndebug2: resolve_canonicalize: hostname *is address\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 1041\r\ndebug3: mux_client_request_session: session request sent\r\n1+3 records in\n1+3 records out\n134675 bytes (135 kB, 132 KiB) copied, 0.00253035 s, 53.2 MB/s\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<NUC> (0, b'', b'OpenSSH_8.2p1 Ubuntu-4ubuntu0.1, OpenSSL 1.1.1f  31 Mar 2020\r\ndebug1: Reading configuration data /home/hawkwood/.ssh/config\r\ndebug1: /home/hawkwood/.ssh/config line 1: Applying options for *\r\ndebug1: /home/hawkwood/.ssh/config line 20: Applying options for NUC\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no files\r\ndebug1: /etc/ssh/ssh_config line 21: Applying options for *\r\ndebug2: resolve_canonicalize: hostname *is address\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 1041\r\ndebug3: mux_client_request_session: session request sent\r\n1+3 records in\n1+3 records out\n134675 bytes (135 kB, 132 KiB) copied, 0.00253035 s, 53.2 MB/s\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<NUC> ESTABLISH SSH CONNECTION FOR USER: hawkwood
<NUC> SSH: ansible.cfg set ssh_args: (-C)(-o)(ControlMaster=auto)(-o)(ControlPersist=60s)
<NUC> SSH: ANSIBLE_HOST_KEY_CHECKING/host_key_checking disabled: (-o)(StrictHostKeyChecking=no)
<NUC> SSH: ANSIBLE_REMOTE_PORT/remote_port/ansible_port set: (-o)(Port=2214)
<NUC> SSH: ANSIBLE_REMOTE_USER/remote_user/ansible_user/user/-u set: (-o)(User="hawkwood")
<NUC> SSH: ANSIBLE_TIMEOUT/timeout set: (-o)(ConnectTimeout=10)
<NUC> SSH: PlayContext set ssh_common_args: ()
<NUC> SSH: PlayContext set ssh_extra_args: ()
<NUC> SSH: found only ControlPersist; added ControlPath: (-o)(ControlPath=/home/hawkwood/.ansible/cp/884cb4555b)
<NUC> SSH: EXEC sshpass -d11 ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=2214 -o 'User="hawkwood"' -o ConnectTimeout=10 -o ControlPath=/home/hawkwood/.ansible/cp/884cb4555b NUC '/bin/sh -c '"'"'chmod u+x /tmp/ansible-tmp-1596602165.7738736-210641032471126/ /tmp/ansible-tmp-1596602165.7738736-210641032471126/AnsiballZ_apt_repository.py && sleep 0'"'"''
<NUC> (0, b'', b'OpenSSH_8.2p1 Ubuntu-4ubuntu0.1, OpenSSL 1.1.1f  31 Mar 2020\r\ndebug1: Reading configuration data /home/hawkwood/.ssh/config\r\ndebug1: /home/hawkwood/.ssh/config line 1: Applying options for *\r\ndebug1: /home/hawkwood/.ssh/config line 20: Applying options for NUC\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no files\r\ndebug1: /etc/ssh/ssh_config line 21: Applying options for *\r\ndebug2: resolve_canonicalize: hostname *is address\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 1041\r\ndebug3: mux_client_request_session: session request sent\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
Escalation succeeded
<NUC> (1, b'\r\n\r\n{"msg": "apt cache update failed", "failed": true, "exception": "  File \\"/tmp/ansible_apt_repository_payload_wr9f_sfa/ansible_apt_repository_payload.zip/ansible/modules/packaging/os/apt_repository.py\\", line 548, in main\\n  File \\"/usr/lib/python3/dist-packages/apt/cache.py\\", line 591, in update\\n    raise FetchFailedException(e)\\n", "invocation": {"module_args": {"filename": "nginx-amplify", "repo": "deb [arch=amd64] http://packages.amplify.nginx.com/ubuntu/ focal amplify-agent", "state": "present", "update_cache": true, "install_python_apt": true, "validate_certs": true, "mode": null, "codename": null}}}\r\n', b'OpenSSH_8.2p1 Ubuntu-4ubuntu0.1, OpenSSL 1.1.1f  31 Mar 2020\r\ndebug1: Reading configuration data /home/hawkwood/.ssh/config\r\ndebug1: /home/hawkwood/.ssh/config line 1: Applying options for *\r\ndebug1: /home/hawkwood/.ssh/config line 20: Applying options for NUC\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no files\r\ndebug1: /etc/ssh/ssh_config line 21: Applying options for *\r\ndebug2: resolve_canonicalize: hostname * is address\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 1041\r\ndebug3: mux_client_request_session: session request sent\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 1\r\nShared connection to *closed.\r\n')
<NUC> Failed to connect to the host via ssh: OpenSSH_8.2p1 Ubuntu-4ubuntu0.1, OpenSSL 1.1.1f  31 Mar 2020
debug1: Reading configuration data /home/hawkwood/.ssh/config
debug1: /home/hawkwood/.ssh/config line 1: Applying options for *
debug1: /home/hawkwood/.ssh/config line 20: Applying options for NUC
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no files
debug1: /etc/ssh/ssh_config line 21: Applying options for *
debug2: resolve_canonicalize: hostname *is address
debug1: auto-mux: Trying existing master
debug2: fd 3 setting O_NONBLOCK
debug2: mux_client_hello_exchange: master version 4
debug3: mux_client_forwards: request forwardings: 0 local, 0 remote
debug3: mux_client_request_session: entering
debug3: mux_client_request_alive: entering
debug3: mux_client_request_alive: done pid = 1041
debug3: mux_client_request_session: session request sent
debug3: mux_client_read_packet: read header failed: Broken pipe
debug2: Received exit status from master 1
Shared connection to *closed.
<NUC> ESTABLISH SSH CONNECTION FOR USER: hawkwood
<NUC> SSH: ansible.cfg set ssh_args: (-C)(-o)(ControlMaster=auto)(-o)(ControlPersist=60s)
<NUC> SSH: ANSIBLE_HOST_KEY_CHECKING/host_key_checking disabled: (-o)(StrictHostKeyChecking=no)
<NUC> SSH: ANSIBLE_REMOTE_PORT/remote_port/ansible_port set: (-o)(Port=2214)
<NUC> SSH: ANSIBLE_REMOTE_USER/remote_user/ansible_user/user/-u set: (-o)(User="hawkwood")
<NUC> SSH: ANSIBLE_TIMEOUT/timeout set: (-o)(ConnectTimeout=10)
<NUC> SSH: PlayContext set ssh_common_args: ()
<NUC> SSH: PlayContext set ssh_extra_args: ()
<NUC> SSH: found only ControlPersist; added ControlPath: (-o)(ControlPath=/home/hawkwood/.ansible/cp/884cb4555b)
<NUC> SSH: EXEC sshpass -d11 ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=2214 -o 'User="hawkwood"' -o ConnectTimeout=10 -o ControlPath=/home/hawkwood/.ansible/cp/884cb4555b NUC '/bin/sh -c '"'"'rm -f -r /tmp/ansible-tmp-1596602165.7738736-210641032471126/ > /dev/null 2>&1 && sleep 0'"'"''
<NUC> (0, b'', b'OpenSSH_8.2p1 Ubuntu-4ubuntu0.1, OpenSSL 1.1.1f  31 Mar 2020\r\ndebug1: Reading configuration data /home/hawkwood/.ssh/config\r\ndebug1: /home/hawkwood/.ssh/config line 1: Applying options for *\r\ndebug1: /home/hawkwood/.ssh/config line 20: Applying options for NUC\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no files\r\ndebug1: /etc/ssh/ssh_config line 21: Applying options for *\r\ndebug2: resolve_canonicalize: hostname *is address\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 1041\r\ndebug3: mux_client_request_session: session request sent\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
The full traceback is:
  File "/tmp/ansible_apt_repository_payload_wr9f_sfa/ansible_apt_repository_payload.zip/ansible/modules/packaging/os/apt_repository.py", line 548, in main
  File "/usr/lib/python3/dist-packages/apt/cache.py", line 591, in update
    raise FetchFailedException(e)
fatal: [nuc]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "codename": null,
            "filename": "nginx-amplify",
            "install_python_apt": true,
            "mode": null,
            "repo": "deb [arch=amd64] http://packages.amplify.nginx.com/ubuntu/ focal amplify-agent",
            "state": "present",
            "update_cache": true,
            "validate_certs": true
        }
    },
    "msg": "apt cache update failed"
}
[WARNING]: Failure using method (v2_runner_on_failed) in callback plugin (<ansible.plugins.callback.mail.CallbackModule
object at 0x7f723879a820>): [Errno 111] Connection refused
Callback Exception:
  File "/usr/lib/python3/dist-packages/ansible/executor/task_queue_manager.py", line 324, in send_callback
    method(*new_args, **kwargs)
   File "/usr/lib/python3/dist-packages/ansible/plugins/callback/mail.py", line 223, in v2_runner_on_failed
    self.mail_result(result, 'Failed')
   File "/usr/lib/python3/dist-packages/ansible/plugins/callback/mail.py", line 213, in mail_result
    self.mail(subject=subject, body=body)
   File "/usr/lib/python3/dist-packages/ansible/plugins/callback/mail.py", line 108, in mail
    smtp = smtplib.SMTP(self.smtphost, port=self.smtpport)
   File "/usr/lib/python3.8/smtplib.py", line 253, in __init__
    (code, msg) = self.connect(host, port)
   File "/usr/lib/python3.8/smtplib.py", line 339, in connect
    self.sock = self._get_socket(host, port, self.timeout)
   File "/usr/lib/python3.8/smtplib.py", line 308, in _get_socket
    return socket.create_connection((host, port), timeout,
   File "/usr/lib/python3.8/socket.py", line 808, in create_connection
    raise err
   File "/usr/lib/python3.8/socket.py", line 796, in create_connection
    sock.connect(sa)
```

**Comment by TimeTravelersHackedMe** [comment]
Update: I got it working by specifying the ansible interpreter as /usr/bin/python2

**Pull request merged**